src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/Body.js

```diff
@@ -1,21 +1,151 @@
 // @flow
 import type {ChildrenArray, Element} from 'react';
 import React from 'react';
+import Icon from '../Icon';
+import Cell from './Cell';
 import Row from './Row';
+import type {ControlItems, ControlConfig, RowProps, SelectMode} from './types';
+import tableStyles from './table.scss';
 
 type Props = {
+    /** Child nodes of the table body */
     children: ChildrenArray<Element<typeof Row>>,
+    /** 
+     * @ignore 
+     * List of buttons to apply action handlers to every row (e.g. edit row) forwarded from table 
+     */
+    controls?: ControlItems,
+    /**
+     * @ignore
+     * Can be set to "single" or "multiple". Defaults is "none".
+     */
+    selectMode?: SelectMode,
+    /** 
+     * @ignore Callback function to notify about selection and deselection of a row
+     */
+    onRowSelectionChange?: (rowId: string | number, selected: boolean) => void,
 };
 
 export default class Body extends React.PureComponent<Props> {
+    static defaultProps = {
+        selectMode: 'none',
+    };
+
+    isSelectable = () => {
+        const {selectMode} = this.props;
+
+        return selectMode === 'single' || selectMode === 'multiple';
+    };
+
+    cloneRows = (originalRows: ChildrenArray<Element<typeof Row>>) => {
+        return React.Children.map(originalRows, (row, index) => {
+            if (React.isValidElement(row)) {
```

Comment:
Verifies the object is a React element. Returns true or false.

https://facebook.github.io/react/docs/react-api.html#isvalidelement